lib/katakata_irb/types.rb

```diff
@@ -1,10 +1,34 @@
 require 'rbs'
 require 'rbs/cli'
+require 'katakata_irb/loader'
 
 module KatakataIrb; end
 module KatakataIrb::Types
   def self.rbs_builder
-    @rbs_builder ||= load_rbs_builder
+    preload if @rbs_builder.nil?
+
+    @rbs_builder
+  end
+
+  def self.rbs_builder=(rbs_builder)
+    @rbs_builder = rbs_builder
+  end
+
+  def self.loader_type=(loader_type)
+    @loader_type = loader_type
+  end
+
+  def self.preload
+    rbs_builder_loader.run if @rbs_builder.nil?
+  end
+
+  def self.rbs_builder_loader
+    case @loader_type
+    when :sync, nil
+      @rbs_builder_loader ||= KatakataIrb::SynchronizedRBSBuilderLoader.new
+    when :async
+      @rbs_builder_loader ||= KatakataIrb::AsyncRBSBuilderLoader.new
```

Comment:
I think this loader pattern is a little heavy.
Could it be simple like this?
```ruby
def self.preload
  return unless @rbs_builder
  ...
  when :async
    @loader_thread ||= Thread.new do
      @rbs_builder = load_rbs_builder
    end
```

Reply:
Done